extensions/sql-database-projects/src/models/publishProfile/publishProfile.ts

```diff
@@ -129,7 +134,151 @@ async function readConnectionString(xmlDoc: any): Promise<{ connectionId: string
+	if ((settings as ISqlProjectPublishSettings).connectionUri) {
+		return true
+	}
+	return false
+}
+
+export function ifIsISqlDbDeployProfile(settings: ISqlProjectPublishSettings | ISqlDbDeployProfile | IPublishToDockerSettings | undefined): settings is ISqlDbDeployProfile {
+	if ((settings as ISqlDbDeployProfile).deploySettings) {
+		return true
+	}
+	return false
+}
+
+export function ifIsIPublishToDockerSettings(settings: ISqlProjectPublishSettings | ISqlDbDeployProfile | IPublishToDockerSettings | undefined): settings is IPublishToDockerSettings {
+	if ((settings as IPublishToDockerSettings).dockerSettings) {
+		return true
+	}
+	return false
+}
+
+export async function getConnectionString(settings: ISqlProjectPublishSettings | ISqlDbDeployProfile | IPublishToDockerSettings | undefined): Promise<string> {
+	let connectionUri: string = '';
+	let connectionString: string = '';
+
+	if (settings) {
+		if (ifIsISqlProjectPublishSettings(settings)) {
+			connectionUri = settings.connectionUri;
+		} else if (ifIsISqlDbDeployProfile(settings)) {
+			connectionUri = settings.deploySettings?.connectionUri ?? '';
+		} else if (ifIsIPublishToDockerSettings(settings)) {
+			connectionUri = settings.sqlProjectPublishSettings.connectionUri;
+		}
+	}
+
+	if (connectionUri) {
+		connectionString = await (await utils.getVscodeMssqlApi()).getConnectionString(connectionUri, false);
+	}
+	return connectionString;
+}
+
+export function getDatabaseName(settings: ISqlProjectPublishSettings | ISqlDbDeployProfile | IPublishToDockerSettings | undefined, projectName: string): string {
+	let databaseName: string = projectName;
+
+	if (settings) {
+		if (ifIsISqlProjectPublishSettings(settings)) {
+			databaseName = settings.databaseName;
+		} else if (ifIsISqlDbDeployProfile(settings)) {
+			databaseName = settings.deploySettings?.databaseName ?? '';
+		} else if (ifIsIPublishToDockerSettings(settings)) {
+			databaseName = settings.sqlProjectPublishSettings.databaseName;
+		}
+	}
+
+	return databaseName;
+}
+
+export async function getDeploymentOptions(settings: ISqlProjectPublishSettings | ISqlDbDeployProfile | IPublishToDockerSettings | undefined, project: Project): Promise<vscodeMssql.DeploymentOptions | undefined> {
+	let deploymentOptions: vscodeMssql.DeploymentOptions | undefined;
+
+	if (settings) {
+		if (ifIsISqlProjectPublishSettings(settings)) {
+			deploymentOptions = settings.deploymentOptions;
+		} else if (ifIsISqlDbDeployProfile(settings)) {
+			deploymentOptions = settings.deploySettings?.deploymentOptions;
+		} else if (ifIsIPublishToDockerSettings(settings)) {
+			deploymentOptions = settings.sqlProjectPublishSettings.deploymentOptions;
+		}
+	} else {
+		deploymentOptions = await utils.getDefaultPublishDeploymentOptions(project);
+	}
+
+	return deploymentOptions;
+}
+
+export function getSqlCmdVariables(settings: ISqlProjectPublishSettings | ISqlDbDeployProfile | IPublishToDockerSettings | undefined): Map<string, string> | undefined {
```

Comment:
nit: if these functions aren't used outside of this file, they don't need to be exported